ntp plugin crash

A parse error in the ntp plugin caused a telegraf agent to crash.

### Relevant telegraf.conf:
```
[[inputs.ntpq]]
#   ## If false, set the -n ntpq flag. Can reduce metric gather time.
#   dns_lookup = true
```
### System info:

Telegraf - version 0.13.1
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.5 LTS"

### Steps to reproduce:

I'm guessing an unexpected response from the call the agent makes to grab NTP info causes the crash to happen.

### Expected behavior:

Agent doesn't crash

### Actual behavior:

Agent Crashes

### Additional info:

```
2016/08/15 09:46:00 ERROR ntpq: parsing int: 77.759
panic: runtime error: index out of range

goroutine 2202595 [running]:
panic(0x107ebc0, 0xc820010080)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/ntpq.(*NTPQ).Gather(0xc8204da660, 0x7f5ecebaa368, 0xc82078c000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/ntpq/ntpq.go:175 +0xc90
github.com/influxdata/telegraf/agent.gatherWithTimeout.func1(0xc820558120, 0xc8201d0c60, 0xc82078c000)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:174 +0x73
created by github.com/influxdata/telegraf/agent.gatherWithTimeout
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:175 +0xe0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -138,6 +138,7 @@ consistent with the behavior of `collection_jitter`.
 - [#1586](https://github.com/influxdata/telegraf/pull/1586): Remove IF NOT EXISTS from influxdb output database creation.
 - [#1600](https://github.com/influxdata/telegraf/issues/1600): Fix quoting with text values in postgresql_extensible plugin.
 - [#1425](https://github.com/influxdata/telegraf/issues/1425): Fix win_perf_counter "index out of range" panic.
+- [#1634](https://github.com/influxdata/telegraf/issues/1634): Fix ntpq panic when field is missing.
 
 ## v0.13.1 [2016-05-24]
 

--- a/plugins/inputs/ntpq/ntpq.go
+++ b/plugins/inputs/ntpq/ntpq.go
@@ -119,7 +119,7 @@ func (n *NTPQ) Gather(acc telegraf.Accumulator) error {
 
 			// Get integer metrics from output
 			for key, index := range intI {
-				if index == -1 {
+				if index == -1 || index >= len(fields) {
 					continue
 				}
 				if fields[index] == "-" {
@@ -169,7 +169,7 @@ func (n *NTPQ) Gather(acc telegraf.Accumulator) error {
 
 			// get float metrics from output
 			for key, index := range floatI {
-				if index == -1 {
+				if index == -1 || index >= len(fields) {
 					continue
 				}
 				if fields[index] == "-" {

--- a/plugins/inputs/ntpq/ntpq_test.go
+++ b/plugins/inputs/ntpq/ntpq_test.go
@@ -41,6 +41,35 @@ func TestSingleNTPQ(t *testing.T) {
 	acc.AssertContainsTaggedFields(t, "ntpq", fields, tags)
 }
 
+func TestMissingJitterField(t *testing.T) {
+	tt := tester{
+		ret: []byte(missingJitterField),
+		err: nil,
+	}
+	n := &NTPQ{
+		runQ: tt.runqTest,
+	}
+
+	acc := testutil.Accumulator{}
+	assert.NoError(t, n.Gather(&acc))
+
+	fields := map[string]interface{}{
+		"when":   int64(101),
+		"poll":   int64(256),
+		"reach":  int64(37),
+		"delay":  float64(51.016),
+		"offset": float64(233.010),
+	}
+	tags := map[string]string{
+		"remote":       "uschi5-ntp-002.",
+		"state_prefix": "*",
+		"refid":        "10.177.80.46",
+		"stratum":      "2",
+		"type":         "u",
+	}
+	acc.AssertContainsTaggedFields(t, "ntpq", fields, tags)
+}
+
 func TestBadIntNTPQ(t *testing.T) {
 	tt := tester{
 		ret: []byte(badIntParseNTPQ),
@@ -379,6 +408,11 @@ func resetVars() {
 var singleNTPQ = `     remote           refid      st t when poll reach   delay   offset  jitter
 ==============================================================================
 *uschi5-ntp-002. 10.177.80.46     2 u  101  256   37   51.016  233.010  17.462
+`
+
+var missingJitterField = `     remote           refid      st t when poll reach   delay   offset  jitter
+==============================================================================
+*uschi5-ntp-002. 10.177.80.46     2 u  101  256   37   51.016  233.010
 `
 
 var badHeaderNTPQ = `remote      refid   foobar t when poll reach   delay   offset  jitter

